### PR TITLE
#1986 remove zero electrolyte concentration cut-off

### LINKED ISSUE #1986: Remove zero-concentration cut-off event from DFN
- add `cutoff_event` kwarg (boolean, default False) to the full electrolyte model
- in `set_events` in the full electrolyte model, don't add the event if `cutoff_event` is `False`
- make `True` for the SPMe (for cut-off, see discussion in #1982 )
- make the cut-off tolerance smaller when `cutoff_event` is True

Easy issue, assuming the models still work without the event

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Removes the "Zero electrolyte concentration cut-off" event.
Adds some bounds to functions so that the solver still works even if we step into electrolyte depletion

Fixes #1986 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/parameters/lithium_ion_parameters.py [hunk 1]
```diff
@@ -683,6 +693,9 @@ def D_dimensional(self, sto, T):
 
     def j0_dimensional(self, c_e, c_s_surf, T):
         """Dimensional exchange-current density [A.m-2]"""
+        # tol = 1e-3
+        # c_e = c_e  # pybamm.maximum(c_e, tol)
+        # c_s_surf = pybamm.maximum(pybamm.minimum(c_s_surf, 1 - tol), tol)
```
Comment (brosaplanella) [inline pybamm/parameters/lithium_ion_parameters.py:698]: Should be deleted

File: pybamm/parameters/lithium_ion_parameters.py [hunk 2]
```diff
@@ -199,11 +199,13 @@ def _set_dimensional_parameters(self):
 
     def D_e_dimensional(self, c_e, T):
         """Dimensional diffusivity in electrolyte"""
+        c_e = pybamm.maximum(c_e, 10)
```
Comment (brosaplanella) [inline pybamm/parameters/lithium_ion_parameters.py:202]: Out of curiosity, which kind of error do you get if you make this smaller?

Related to that, do we want to hard-code this or set some tolerances that the user can change?

**Review by brosaplanella — APPROVED** [review]
Looks good, thanks!

File: pybamm/parameters/lithium_ion_parameters.py (hunk 2 above)
Comment (tinosulzer) [reply pybamm/parameters/lithium_ion_parameters.py:202]: Making this smaller gives more casadi jacobian warnings. Yes, I will make it into a setting

**Pull request merged**